platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,208 @@
+
+Additionally, we have the preview server which shows content editors any accessibility violations, but they must deliberately go to the server from a link in Drupal to see this.
+
+#### Pain points
+
+##### Content Writers
+
+Content writers want to quickly draft, publish, and deploy content. However, `vets-website` is only deployed to production once per day. To allow content writers to move faster, there is a [partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#partial-deploy--static-page-changes-only), which only includes static page changes (`vagov-content` and Drupal).
+
+Unfortunately, the content-only deploys have no accessibility checking. That means that content writers don't have a proactive way to learn if their content includes accessibility errors. Adding accessibility checking to the CMS is on the roadmap, but it doesn't currently exist.  
+
+##### Users
+
+To enable a faster publishing workflow, all tests (except link checking) are intentionally skipped during content-only deploys. That creates an opportunity for end users to encounter accessibility errors. 
+
+##### FE Tools team
+
+Invalid content causes warning logs during staging deploys, and breaks the build during production deploys. These broken builds then need manual intervention, and require coordinating with another team.
+
+##### Anybody that is trying to ship code for the VA.gov front end
+
+If invalid content is breaking the production build, and therefore blocking the production deploy, then no team on any contract is able to ship their frontend code to VA.gov.
+
+### High Level Design
+
+Once the `content-build` repo is successfully extracted from `vets-website`, the Metalsmith static content build pipeline will be deleted from `vets-website`. Webpack will handle the application build in `vets-website`, and Metalsmith will handle the static content build in the `content-build` repo. This will create two separate pipelines for built code to be deployed to S3, which will prevent failures in one from blocking the other.
+
+The validation of the static content coming from the CMS will occur during the Metalsmith build.
+
+## Specifics
+
+### Detailed Design
+
+The content validation will happen in a daily scheduled job.
+
+### Code Location
+
+The content validation script and scheduled job config will live in the [new `content-build` repo](https://github.com/department-of-veterans-affairs/content-build).
+
+### Testing Plan
+
+1. Configure a scheduled job to run a script once per workday
+2. Write an initial script that logs when the script is run
+3. Once the scheduled job is running at the correct time, replace the initial script with the actual script that runs the content validation
+
+### Logging
+
+Currently, invalid content is logged during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
```

Comment:
Do you plan on changing this?

Reply:
The new content validation script will continue including logs in the build log. On top of that, it will send an automated report of invalid content to the #cms-team Slack channel.